source/dofs/dof_renumbering.cc

```diff
@@ -2248,6 +2254,146 @@ namespace DoFRenumbering
            ExcInternalError());
   }
 
+
+
+  template <int dim, int spacedim>
+  void
+  support_point_wise(DoFHandler<dim, spacedim> &dof_handler)
+  {
+    std::vector<types::global_dof_index> renumbering(
+      dof_handler.n_locally_owned_dofs(), numbers::invalid_dof_index);
+    compute_support_point_wise(renumbering, dof_handler);
+
+    // If there is only one component then there is nothing to do, so check
+    // first:
+    if (Utilities::MPI::max(renumbering.size(),
+                            dof_handler.get_communicator()) > 0)
+      dof_handler.renumber_dofs(renumbering);
+  }
+
+
+
+  template <int dim, int spacedim>
+  void
+  compute_support_point_wise(
+    std::vector<types::global_dof_index> &new_dof_indices,
+    const DoFHandler<dim, spacedim> &     dof_handler)
+  {
+    const types::global_dof_index n_dofs = dof_handler.n_locally_owned_dofs();
+    Assert(new_dof_indices.size() == n_dofs,
+           ExcDimensionMismatch(new_dof_indices.size(), n_dofs));
+
+    // This renumbering occurs in three steps:
+    // 1. Compute the component-wise renumbering so that all DoFs of component
+    //    i are less than the DoFs of component i + 1.
+    // 2. Compute a second renumbering component_to_nodal in which the
+    //    renumbering is now, for two components, [u0, v0, u1, v1, ...]: i.e.,
+    //    DoFs are first sorted by component and then by support point.
+    // 3. Compose the two renumberings to obtain the final result.
+
+    // Step 1:
+    std::vector<types::global_dof_index> component_renumbering(
+      n_dofs, numbers::invalid_dof_index);
+    compute_component_wise<dim, spacedim>(component_renumbering,
+                                          dof_handler.begin_active(),
+                                          dof_handler.end(),
+                                          std::vector<unsigned int>(),
+                                          false);
+
+    const std::vector<types::global_dof_index> dofs_per_component =
+      DoFTools::count_dofs_per_fe_component(dof_handler, true);
+
+    // At this point we have no more communication to do - simplify things by
+    // returning early if possible
+    if (component_renumbering.size() == 0)
+      {
+        new_dof_indices.resize(0);
+        return;
+      }
```

Comment:
Is this going to work in parallel?

Reply:
Yes. I have tests for this case (both with no renumbering and when a processor has no dofs and there is a renumbering).